File: traversal/post-order/recursion.go
package postorder

import binarytree "algorithms/data-structures/binary-tree"

// Recursive for binary tree post-order traversal
func Recursive(root *binarytree.Node) []int {
	answer := []int{}
	traversal(root, &answer)
	return answer
}

func traversal(root *binarytree.Node, answer *[]int) {
	if root == nil {
		return
	}
	traversal(root.Left, answer)
	traversal(root.Right, answer)
	*answer = append(*answer, root.Val.(int))
}

File: traversal/post-order/recursion.ts
import { BinaryTreeNode } from 'data-structures/src/binary_tree'

export function postOrderTraversal<V>(node: BinaryTreeNode<V> | null) {
  const answer: V[] = []
  traversal(node, answer)
  return answer
}

function traversal<V>(node: BinaryTreeNode<V> | null, answer: V[]) {
  if (node === null) return

  traversal(node.left, answer)
  traversal(node.right, answer)
  answer.push(node.value)
}
